tests/test_discriminated_union.py

```diff
@@ -1868,8 +1868,6 @@ class LeafState(BaseModel):
         state_type: Literal['leaf']
 
     AnyState = Annotated[Union[NestedState, LoopState, LeafState], Field(..., discriminator='state_type')]
-    NestedState.model_rebuild()
```

Comment:
If you expand this section, you can see that this test showcases the example I've shown in the PR description 👍